src/main/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolder.java

```diff
@@ -271,6 +281,31 @@ final void updateChildren(final TaskListener listener) throws IOException, Inter
     }
 
     /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void delete() throws IOException, InterruptedException {
+        checkPermission(DELETE);
+        FolderComputation<I> computation = getComputation();
+        Executor executor = Executor.of(computation);
+        if (executor != null) {
+            LOGGER.log(Level.INFO, "Interrupting {0} in order to delete it", this);
+            executor.interrupt(Result.ABORTED, new CauseOfInterruption.UserInterruption(User.current()));
+            // give it 15 seconds or so to respond to the interrupt
+            long expiration = System.nanoTime() + TimeUnit.SECONDS.toNanos(15);
+            while (executor.isAlive()
+                    && executor.getCurrentExecutable() == computation
```

Comment:
Superfluous check I think—it has been a long time since `Executor`s were reused.